tutorials/2023-07-05-neo3-boa-writing-a-fungible-token-contract-using-python/index.md

```diff
@@ -0,0 +1,341 @@
+
+    return True
+```
+
+## 8. NEP-17 Event
+
+The NEP-17 standard defines a single event called `Transfer` that must be triggered when a transfer occurs.
+You can use the `Nep17TransferEvent` from Neo3-boa or create your own event with `CreateNewEvent`.
+
+In this example, we use this event twice: when minting all the tokens at the deploy and whenever a transfer operation occurs.
+
+```python
+# update coin.py adding the following code:
+from boa3.builtin.compile_time import CreateNewEvent
+
+on_nep17_transfer = CreateNewEvent(
+    [
+        ('from', Union[UInt160, None]),
+        ('to', Union[UInt160, None]),
+        ('amount', int),
+    ],
+    'Transfer'
+)
+
+# to trigger the event, just call it by passing the parameters
+# using the already implemented Event:
+Nep17TransferEvent(from_address, to_address, amount)
+# using the newly created Event:
+on_nep17_transfer(from_address, to_address, amount)
+```
+
+## 9. NEP-17 Callbacks
+
+The NEP-17 standard defines a single callback called `onNEP17Payment` that must be called if the recipient is a contract.
+This callback is used to notify the recipient that it has received tokens. It's up to the recipient to implement this
+callback. The recipient can reject the transfer by raising an exception.
+
+```python
+# update coin.py adding the following code:
+from boa3.builtin.contract import abort
+
+@public(name='onNEP17Payment')
+def on_nep17_payment(from_address: UInt160, amount: int, data: Any):
+    abort()     # in this example, the smart contract is rejecting all transfers made to it
+```
+
+## 10. Manifest Metadata
+
+Neo3-boa allows you to define the contract metadata using the `metadata` decorator. This information is used to generate
+the contract manifest file.
+
+In our example, we will define the contract name and the supported standards, but you can add any information you want
+to. We are also adding permissions to allow our contract to call the `onNEP17Payment` method. The compiler automatically
+does this step, but it's a good practice to define the permissions in the contract code.
+
+> You should add this method at the beginning of your file, before any other method.
+
+```python
+from boa3.builtin.compile_time import metadata, NeoMetadata
+
+@metadata
+def manifest_metadata() -> NeoMetadata:
+    meta = NeoMetadata()
+    meta.name = 'Coin Contract'
+    meta.supported_standards = ['NEP-17']
+    meta.add_permission(methods=['onNEP17Payment'])
+    return meta
+```
+
+## 11. Other examples
+
+Check out this [simple NEP-17](https://github.com/CityOfZion/neo3-boa/blob/master/boa3_test/examples/simple_nep17.py)
+example to see a smart contract that is as simple as the one in this example, but is more cohesive and uses some more
+features in its implementation.
+
+Also, check out this [more complex NEP-17](https://github.com/CityOfZion/neo3-boa/blob/master/boa3_test/examples/nep17.py)
+example to examine a smart contract that can also mint and burn tokens.
```

Comment:
since the tutorial uses v1.0.0, it would be good to link to the tag version, since master may be updated (and then be incompatible with the current tutorial)
```suggestion
Check out this [simple NEP-17](https://github.com/CityOfZion/neo3-boa/blob/v1.0.0/boa3_test/examples/simple_nep17.py)
example to see a smart contract that is as simple as the one in this example, but is more cohesive and uses some more
features in its implementation.

Also, check out this [more complex NEP-17](https://github.com/CityOfZion/neo3-boa/blob/v1.0.0/boa3_test/examples/nep17.py)
example to examine a smart contract that can also mint and burn tokens.
```

Note: the link currently doesn't exist since the release is not up yet